_posts/2023-11-07-sssd-authentication-method-policy.md

```diff
@@ -0,0 +1,48 @@
+---
+layout:     post
+title:      "SSSD: authentication method policy"
+date:       2023-11-07 12:00:00 +0100
+categories: idm
+---
+
+# Introduction
+
+SSSD has recently released version [2.9.2](https://sssd.io/release-notes/sssd-2.9.2.html), which adds a new option to control the authentication method policy. The need for this new option comes from adding new authentication methods such as passkey[1][2]. It is possible that more authentication methods will emerge in the future, so it seems necessary to add a way to control the policy governing them.
+

```

Comment:
Hi,

I think it is not only about adding new authentication methods but adding methods which might or might not work with the configured online/server-side authentication and can securely used for local/offline authentication without interacting with the server.

E.g. passkey with an Active Directory client. Currently online authentication with passkey against AD would not work in the sense that we will not get a Kerberos ticket. But if the required attributes are added to AD and read by SSSD a passkey can be used for local authentication if no Kerberos ticket is needed.

Since you mentioned below that `cache_credentials = True` might allow some types of attacks it might be worth to consider using `local_auth_policy = enable:passkey` so that when online password based authentication can be user to get a Kerberos ticket and if offline the passkey has to be used. In this setup the passkey can be used when being online as well but then you won't get a Kerberos ticket. Btw, maybe it is worth to mention that there will be a message in this case telling you that there will be no Kerberos ticket.